src/extra/flattenConcurrentlyAtMost.ts

```diff
@@ -0,0 +1,132 @@
+import {Operator, Stream, OutSender, InternalListener} from '../index';
+
+class FCAMIL<T> implements InternalListener<T>, OutSender<T> {
+  constructor(public out: Stream<T>,
+              private op: FlattenConcAMOperator<T>) {
+  }
+
+  _n(t: T) {
+    this.out._n(t);
+  }
+
+  _e(err: any) {
+    this.out._e(err);
+  }
+
+  _c() {
+    this.op.less();
+  }
+}
+
+export class FlattenConcAMOperator<T> implements Operator<Stream<T>, T> {
+  public type = 'flattenConcurrentlyAtMost';
+  public out: Stream<T> = null as any;
+  private _l: number = 0;
+  private _d: boolean = false;
+  private _seq: Array<Stream<T>> = [];
+
+  constructor(public n: number, public ins: Stream<Stream<T>>) {
+  }
+
+  _start(out: Stream<T>): void {
+    this.out = out;
+    this.ins._add(this);
+  }
+
+  _stop(): void {
+    this.ins._remove(this);
+    this._l = 0;
+    this.out = null as any;
+    this._seq = [];
+  }
+
+  less(): void {
+    const seq = this._seq;
+    if (--this._l === 0 && seq.length === 0 && this._d) {
+      const u = this.out;
+      if (!u) return;
+      u._c();
+    }
+    if (this._l === 0 && seq.length > 0) {
```

Comment:
Why do we need this `this._l === 0` check? Suppose `n === 3` and `this._l === 3`, once this `less()` is called, now `this._l === 2` and `2 < 3` so we should try to get stuff from the seq.

Reply:
Good catch, that's a bug. We don't want to exhaust the list before draining the queue, I'll fix.